aesm-client/src/imp/aesm_protobuf/mod.rs

```diff
@@ -0,0 +1,114 @@
+use imp::AesmClient;
+pub use error::{AesmError, Error, Result};
+use protobuf::Message;
+use std::io::{Read, Write};
+use byteorder::{LittleEndian, NativeEndian, ReadBytesExt, WriteBytesExt};
+use {
+    AesmRequest, FromResponse, QuoteInfo, QuoteResult, QuoteType,
+    Request_GetQuoteRequest, Request_InitQuoteRequest,
+};
+// FIXME: remove conditional compilation after resolving https://github.com/fortanix/rust-sgx/issues/31
+#[cfg(not(target_env = "sgx"))]
+use imp::{LOCAL_AESM_TIMEOUT_US, REMOTE_AESM_TIMEOUT_US};
+// FIXME: remove conditional compilation after resolving https://github.com/fortanix/rust-sgx/issues/31
+#[cfg(not(target_env = "sgx"))]
+use std::time::Duration;
+
+impl AesmClient {
+    pub fn try_connect(&self) -> Result<()> {
+        self.open_socket().map(|_| ())
+    }
+
+    pub(super) fn transact<T: AesmRequest>(&self, req: T) -> Result<T::Response> {
+        let mut sock = self.open_socket()?;
+
+        // FIXME: remove conditional compilation after resolving https://github.com/fortanix/rust-sgx/issues/31
+        #[cfg(not(target_env = "sgx"))]
+        let _ = sock.set_read_timeout(req.get_timeout().map(|t| Duration::from_micros(t as _)))?;
+
+        let req_bytes = req
+            .into()
+            .write_to_bytes()
+            .expect("Failed to serialize protobuf");
+        sock.write_u32::<NativeEndian>(req_bytes.len() as u32)?;
+        sock.write_all(&req_bytes)?;
+
+        let res_len = sock.read_u32::<NativeEndian>()?;
+        let mut res_bytes = vec![0; res_len as usize];
+        sock.read_exact(&mut res_bytes)?;
+
+        let res = T::Response::from_response(protobuf::parse_from_bytes(&res_bytes))?;
+        Ok(res)
+    }
+
+    // FIXME: remove conditional compilation after resolving https://github.com/fortanix/rust-sgx/issues/31
+    #[cfg(not(target_env = "sgx"))]
```

Comment:
Just do this conditional inline (as in `get_quote`). You can apply `#[allow(unused_mut)]` on the binding.